pkg/datapath/linux/routing/eni.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package linuxrouting
+
+import (
+	"net"
+
+	"github.com/cilium/cilium/pkg/datapath/linux/linux_defaults"
+	"github.com/cilium/cilium/pkg/datapath/linux/route"
+
+	"github.com/vishvananda/netlink"
+)
+
+// ENIRulesAndRoutesOptions are options for ENIRulesAndRoutes.
+type ENIRulesAndRoutesOptions struct {
+	EgressMultiHomeIPRuleCompat bool
+	EnableIPv4Masquerade        bool
+}
+
+// ENIRulesAndRoutes returns the rules and routes required to configure an ENI
+// interface.
+func ENIRulesAndRoutes(
```

Comment:
So this logic is not ENI specific anymore, right? It's basically just extracting the logic from `routingInfo.Configure` into a side-effect free function. We should probably call this something like `ComputeRulesAndRoutes` or something.